saml/src/test/java/com/linecorp/armeria/saml/SamlServiceProviderConfiguratorTest.java

```diff
@@ -0,0 +1,546 @@
+            idpKeyStore.load(null, null);
+
+            final SelfSignedCertificate idp = new SelfSignedCertificate();
+            idpKeyStore.setKeyEntry("signing", idp.key(), "".toCharArray(),
+                                    new Certificate[] { idp.cert() });
+            final CredentialResolver idpCredentialResolver =
+                    new KeyStoreCredentialResolver(idpKeyStore, ImmutableMap.of("signing", ""));
+
+            final CriteriaSet cs = new CriteriaSet();
+            cs.add(new EntityIdCriterion("signing"));
+            idpCredential = idpCredentialResolver.resolveSingle(cs);
+
+            // Create my key store for testing.
+            final KeyStore myKeyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            myKeyStore.load(null, null);
+
+            final SelfSignedCertificate mine = new SelfSignedCertificate();
+            // Add my keys for signing and encryption.
+            myKeyStore.setKeyEntry("signing", mine.key(), "".toCharArray(),
+                                   new Certificate[] { mine.cert() });
+            myKeyStore.setKeyEntry("encryption", mine.key(), "".toCharArray(),
+                                   new Certificate[] { mine.cert() });
+
+            // Add IdPs' certificates for validating a SAML message from the IdP.
+            // By default, Armeria finds the certificate whose name equals to the entity ID of an IdP,
+            // so we are adding the certificate with IdP's entity ID.
+            myKeyStore.setCertificateEntry("http://idp.example.com/post", idp.cert());
+            myKeyStore.setCertificateEntry("http://idp.example.com/redirect", idp.cert());
+
+            // Create a password map for my keys.
+            final Map<String, String> myKeyPasswords = ImmutableMap.of("signing", "",
+                                                                       "encryption", "");
+            myCredentialResolver = new KeyStoreCredentialResolver(myKeyStore, myKeyPasswords);
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+    }
+
+    @ClassRule
+    public static ServerRule rule = new ServerRule() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            try (ServerSocket ss = new ServerSocket()) {
+                ss.bind(new InetSocketAddress("127.0.0.1", 0));
+                final int port = ss.getLocalPort();
+                sb.port(port, SessionProtocol.HTTP);
+
+                final SamlServiceProviderConfigurator authConfigurator = SamlServiceProviderConfigurator
+                        .builder()
+                        // A request will be authenticated if it contains 'test=test' cookie in Cookie header.
+                        .addAuthorizer(new CookieBasedAuthorizer("test", "test"))
+                        .authManager(new CookieBasedAuthManager("test", "test"))
+                        // My entity ID
+                        .entityId("http://127.0.0.1:" + port)
```

Comment:
All these URLs could be auto-filled:

- scheme could be detected automatically between `http` and `https` from the active `ServerPort`s.
- hostname could be detected automatically from the `VirtualHost` where the SAML service/decorator is added.
- port could be detected automatically from the active `ServerPort`s.
- All path could have their own default values so that a user doesn't have to specify them.
- Both scheme and hostname could be overridden if a user does not like the auto-detected values.